File: GFG/Floyd Warshall/Floyd Warshall.java
class Solution
{
    public void shortest_distance(int[][] mat)
    {
        // Code here 
        int n=mat.length;
        for(int k=0;k<n;k++){
            for(int i=0;i<n;i++){
                for(int j=0;j<n;j++){
                    if(mat[i][k]==-1 || mat[k][j]==-1)continue;
                    if(mat[i][j]==-1 || mat[i][k]+mat[k][j]<mat[i][j]){
                        mat[i][j]=mat[i][k]+mat[k][j];
                    }
                }
            }
        }
    }
}

File: GFG/Floyd Warshall/FloydWarshall.cpp
class Solution {
  public:
 void shortest_distance(vector<vector<int>>&mat)
     int n = mat.size();
     for(int i = 0;i<n;i++)
     {
         for(int j = 0;j<n;j++)
         {
             for(int k = 0;k<n;k++)
             {
                 if(mat[j][k] == -1)
                 {
                     if(mat[j][i] != -1 && mat[i][k] != -1)
                     mat[j][k] = mat[j][i] + mat[i][k];
                 }
                 else 
                {
                     if(mat[j][i] != -1 && mat[i][k] != -1)
                     mat[j][k] = min(mat[j][k] , mat[j][i]+mat[i][k]);
                }              
             }
         }
     }
 }
};
